Test Ant saxon.custom.options property

AirQuick opened the pull request:
This pull request adds tests for Ant `saxon.custom.options` property both for XSLT and XQuery.

The property is not documented in Wiki, but it's been in `build.xml` for some years. And we know its use case (#235).
So it should be worth testing.

---

This pull request derives from #266 . So needs to be handled after that.

AirQuick commented:
The latest commit's automated test logs: [Travis](https://travis-ci.org/AirQuick/xspec/builds/444519562), [AppVeyor](https://ci.appveyor.com/project/AirQuick/xspec/builds/19680555)

cirulls commented:
I redelivered the payload to GitHub today (once the [GitHub's outage](https://blog.github.com/2018-10-21-october21-incident-report/) was fully solved) and checks are all green now. I should be able to review this in the next days.

cirulls commented on test/saxon-custom-options/config.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration edition="HE" xmlns="http://saxon.sf.net/ns/configuration">
+	<collations>
+		<collation ignore-case="yes" lang="en" uri="x-urn:test:saxon-custon-options:collation" />
```
I think there is a typo here, it should be:

```
uri="x-urn:test:saxon-custom-options:collation"
```

Note `custom` instead of `custon` (which I believe is a typo).

cirulls commented on test/saxon-custom-options/test.xspec:
```diff
@@ -0,0 +1,13 @@
+		<x:call function="compare">
+			<x:param select="'a'" />
+			<x:param select="'A'" />
+			<x:param select="'x-urn:test:saxon-custon-options:collation'" />
```
I think there is a typo here, it should be:

```
uri="x-urn:test:saxon-custom-options:collation"
```

Note `custom` instead of `custon` (which I believe is a typo).

AirQuick commented:
Thanks for spotting it. Fixed now.

> I think it would be good to document the support for Saxon custom options in the wiki. 

Done for [Ant](https://github.com/xspec/xspec/wiki/Running-with-Ant#useful-properties).
And updated a comment in `build.xml` to reflect it. (https://github.com/xspec/xspec/pull/267/commits/5ef29d92dbd7e6de8fd7037badf99f701f3471d9)
